api/envoy/service/mutation/v3alpha/mutation.proto

```diff
@@ -0,0 +1,302 @@
+
+    SendBufferedResponseBodyCommand send_buffered_response_body = 7;
+
+    SendRequestTrailersCommand send_request_trailers = 8;
+
+    SendResponseTrailersCommand send_response_trailers = 9;
+  }
+
+  // Replace the entire HTTP response (whether on the request or on the
+  // response path) with the contents of the CompleteResponse message.
+  // If a StopCommand command is set, then send this response back immediately
+  // to the client with no additional filter processing.
+  CompleteResponse complete_response = 10;
+
+  // Add or replace HTTP headers. This will be ignored after SendXXXBodyChunkCommand has
+  // been sent. If SendXXXTrailers was previously sent, then
+  // modify the trailers instead.
+  repeated config.core.v3.HeaderValueOption set_headers = 11;
+
+  // Remove these HTTP headers. This will be ignored after SendXXXBodyChunk
+  // has been sent. If SendXXXTrailers was previously sent, then
+  // modify the trailers instead.
+  repeated string remove_headers = 12;
+
+  // Replace some or all of the last message body received with a new
+  // set of bytes. This will be ignored unless the previous message sent
+  // by the client is a "Body" request.
+  BodyMutation body_mutation = 13;
+
+  // Add new metadata to the Envoy context
+  google.protobuf.Struct new_metadata = 14;
+
+  // Change how and if the filter waits for responses from the server
+  ProcessingModeChange async_mode_change = 15;
+}
+
+// This message is sent to the mutation server when the HTTP request is
+// first received.
+// [#next-free-field: 9]
+message Request {
+  // A unique ID that will match the one sent in the "Response" message
+  string id = 1;
+
+  // The HTTP request method, such as `GET`, `POST`.
+  string method = 2;
+
+  // The HTTP request headers. If multiple headers share the same key, they
+  // must be merged according to the HTTP spec. All header keys must be
+  // lower-cased, because HTTP header keys are case-insensitive.
+  config.core.v3.HeaderMap headers = 3;
+
+  // The request target, as it appears in the first line of the HTTP request. This includes
+  // the URL path and query-string. No decoding is performed.
+  string path = 4;
+
+  // The HTTP request `Host` or 'Authority` header value.
+  string host = 5;
+
+  // The HTTP URL scheme, such as `http` and `https`.
+  string scheme = 6;
+
+  // Information about the downstream client's connection
+  auth.v3.AttributeContext.Peer client_info = 7;
+
+  // The values of properties selected by the "request_properties"
+  // list in the configuration. Each entry in the map is the
+  // value of the specified property.
+  map<string, google.protobuf.Struct> properties = 8;
+}
+
+// This message is sent to the mutation server when the HTTP response is first
+// received.
+// [#next-free-field: 6]
+message Response {
+  // A unique ID that will match the one sent in the "Request" message
+  string id = 1;
+
+  // The HTTP status on the response
+  int32 status_code = 2;
```

Comment:
redundant

Reply:
Sure, but useful? There are similar redundant fields in the request message that I copied from the ext_authz filter.